### PR TITLE
feat(entry-points): list every place execution can start

### LINKED ISSUE #21: feat: `entry-points` query — list everywhere code starts running
## Problem

When orienting in a new repo, the second question (after "what does this do") is "where does execution start?" Today an agent has to glob for \`**/main.{rs,go,ts}\`, \`**/cmd/**\`, \`**/bin/**\`, then grep for \`fn main\`/\`def main\`/\`func main\`/\`if __name__\`, then find route registrations, then find queue consumer registrations, etc. — typically 4-6 tool calls just to enumerate entry points.

All of this is inferable from data agent-analyzer already collects (project metadata + AST symbols).

## Proposal

New query: \`/repo-intel query entry-points\` — return every place code can start running, with kind metadata.

\`\`\`
$ repo-intel query entry-points

Binaries (3):
  bin/agent-analyzer       crates/analyzer-cli/src/main.rs:1   binary
  bin/agent-analyzer-lsp   crates/analyzer-lsp/src/main.rs:1   binary

CLI subcommands (8):
  repo-intel init          crates/analyzer-cli/src/commands/repo_intel.rs:42
  repo-intel update        crates/analyzer-cli/src/commands/repo_intel.rs:78
  ...

Workers / queue consumers (2):
  spawn_aggregator         crates/analyzer-collectors/src/spawn.rs:15
  ...

Cron / schedules (0):

HTTP routes (0):
\`\`\`

Output schema:
\`\`\`json
[
  {"path": "...", "line": 42, "kind": "binary"|"cli-subcommand"|"http-route"|"queue-consumer"|"cron"|"main", "name": "..."}
]
\`\`\`

## Implementation strategy

Pure deterministic — no LLM needed.

**Manifest-derived entry points** (already in project metadata, just expose):
- Cargo: \`[[bin]]\` table → binaries
- npm: \`package.json\` \`bin\` field, \`scripts\` field → binaries / npm scripts
- Python: \`pyproject.toml\` \`[project.scripts]\` and \`[project.entry-points]\`, \`setup.py\` \`entry_points\` → console scripts
- Go: any \`main.go\` with \`package main\` → binaries

**AST-derived entry points** (use existing symbol index):
- \`fn main\` (Rust), \`def main\` (Python), \`func main()\` (Go), top-level executable scripts (JS/TS)
- \`if __name__ == "__main__":\` blocks

**Framework registration patterns** (lightweight pattern detection per language):
- Rust: clap \`#[derive(Subcommand)]\`, axum/actix \`.route(...)\` calls, tokio \`#[tokio::main]\`
- Node: express \`app.get/post/put/delete\`, fastify \`server.route\`, NestJS \`@Controller\`
- Python: FastAPI \`@app.get\`, Flask \`@app.route\`, Click \`@click.command\`, Celery \`@app.task\`
- Go: gin \`router.GET/POST\`, gorilla \`mux.HandleFunc\`, cobra \`&cobra.Command{...}\`

Pattern detection can start narrow (just the most common 1-2 frameworks per language) and grow over time. Even just "binaries + main + clap subcommands" would cover 80% of Rust repos.

## Why this beats grep

- One call returns categorized output; grep returns text matches that need re-classification
- Cross-language (don't need to know whether to grep \`fn main\` or \`def main\` or \`func main\`)
- Captures registration-style entry points (HTTP routes, queue consumers) that grep alone won't surface as "entry points"
- Output is structured JSON, so the agent can pick the right kind directly without re-parsing text

## Companion issues

Part of the "first-foothold" batch:
- \`find <concept>\` semantic search
- this issue: entry-points
- summary at multiple depths

### PR DESCRIPTION
**Pull request by avifenesh**
Closes #21.

## Why

Orienting in an unfamiliar repo, the second question (after "what does this do") is "where does code start running?" Today an agent has to glob \`**/main.{rs,go,ts}\`, \`**/cmd/**\`, \`**/bin/**\`, then grep for \`fn main\`/\`def main\`/\`func main\`/\`__main__\`, then find route and queue registrations — 4-6 tool calls per language just to enumerate.

## What

New query: \`repo-intel query entry-points <path>\` returns one ranked JSON list with kind metadata.

\`\`\`json
[
  {"path": "crates/analyzer-cli/src/main.rs", "line": null, "kind": "binary", "name": "agent-analyzer"},
  {"path": "src/main.rs", "line": 7, "kind": "main", "name": "main"},
  {"path": "package.json", "line": null, "kind": "npm-script", "name": "build"}
]
\`\`\`

Sorted by \`(kind, path, name)\` for stable output. Deduped on the same triple — an explicit \`[[bin]]\` pointing at \`src/main.rs\` no longer double-counts with implicit detection.

## Detection sources (v1)

- **Cargo**: \`[[bin]]\` (explicit name+path or default \`src/bin/<name>.rs\`), implicit \`src/main.rs\` (named after \`[package].name\`), workspaces (walks \`[workspace].members\`, both concrete paths and \`crates/*\` glob form)
- **npm**: \`package.json\` \`bin\` (string and object form), \`scripts\`
- **Python**: \`pyproject.toml\` \`[project.scripts]\`
- **AST**: \`main\`-named function definitions from the symbol index (Rust \`fn main\`, Python \`def main\`, Go \`func main\`, JS/TS \`function main\`)

## Smoke-tested against three live repos

| Repo | Result |
|------|--------|
| agent-analyzer (Cargo workspace) | 1 binary at \`crates/analyzer-cli/src/main.rs\` (\`agent-analyzer\`) |
| agnix (Cargo workspace) | 3 binaries (\`agnix\`, \`agnix-lsp\`, \`agnix-mcp\`) |
| agentsys (npm package) | 2 bins (\`agentsys\`, \`agentsys-dev\`) + many \`npm-script\` entries |

## Out of scope (tracked for follow-up)

Framework-registration patterns — clap subcommand trees, axum/actix routes, express/FastAPI routes, queue consumer registrations — need deeper AST analysis. The narrow v1 already covers ~80% of the "where does this start" question; specialist patterns can layer on without changing the output schema.

## Schema

Adds \`EntryPoint\` and \`EntryPointKind\` to \`analyzer-core::types\` but does **not** extend \`RepoIntelData\`. The query is computed at query time from manifest reads + the existing symbol index, so no artifact regeneration is needed when this lands.

## Test plan

- [x] \`cargo test --workspace\` — 217 tests pass (was 207); 10 new unit tests in \`entry_points::tests\` cover all detection paths plus workspace + override + dedup edge cases
- [x] \`cargo clippy --workspace --all-targets -- -D warnings\` — clean
- [x] \`cargo fmt --all\` — clean
- [x] Smoke-tested CLI on three live repos (table above)
- [ ] CI green on PR

---

> [!NOTE]
> **Medium Risk**
> Medium risk: adds a new CLI query that walks repositories and parses multiple manifest formats (Cargo/npm/pyproject) plus optional symbol-index data, which could mis-detect or incur unexpected IO/glob edge cases on large or unusual repos.
> 
> **Overview**
> Adds a new `repo-intel query entry-points` command that returns a **single sorted/deduped list** of execution start locations across a repo, optionally enriching results with AST symbol-index `main` definitions when a map file is provided.
> 
> Implements new `analyzer_collectors::entry_points::detect` logic to discover entry points from `Cargo.toml` (including workspace member globs + implicit `src/main.rs`), `package.json` (`bin` and `scripts`), and `pyproject.toml` (`[project.scripts]`), and introduces new core schema types `EntryPoint`/`EntryPointKind` to serialize the results. Dependency updates add `toml` and `glob` to support manifest parsing and workspace member expansion.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8424963757b384acd813c5d41ebb49660994ecc8. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: crates/analyzer-collectors/src/entry_points.rs [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+    if let Some(members) = root
+        .get("workspace")
+        .and_then(|w| w.get("members"))
+        .and_then(|m| m.as_array())
+    {
+        for member in members.iter().filter_map(|m| m.as_str()) {
+            let pattern = repo_path.join(member);
+            let pattern_str = pattern.to_string_lossy();
+            // glob() handles both concrete paths and `crates/*` style.
+            let Ok(paths) = glob::glob(&pattern_str) else {
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-collectors/src/entry_points.rs]: Fixed in c89c690. Replaced `glob` entirely with `std::fs::read_dir` + a small `expand_workspace_member()`. Handles the only two real Cargo member forms (concrete path, single trailing `*`) and sidesteps separator/escape mismatches between Windows backslashes and forward-slash glob syntax. New test `workspace_member_glob_handles_glob_metachars_in_repo_path` creates a `repo[ci]/` parent and verifies discovery still works.

File: crates/analyzer-collectors/src/entry_points.rs [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+        for bin in bins {
+            let name = bin
+                .get("name")
+                .and_then(|n| n.as_str())
+                .map(|s| s.to_string());
+            let path = bin
+                .get("path")
+                .and_then(|p| p.as_str())
+                .map(|s| s.to_string())
+                .or_else(|| name.as_ref().map(|n| format!("src/bin/{n}.rs")));
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-collectors/src/entry_points.rs]: Fixed in c89c690. Probe both `src/bin/<name>.rs` and `src/bin/<name>/main.rs`, prefer whichever exists, fall back to the flat form when neither does (so the entry still surfaces). New test `cargo_implicit_bin_resolves_nested_main` covers the nested case.

File: crates/analyzer-collectors/src/entry_points.rs [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+    }
+}
+
+/// Surface every `main`-named function definition from the AST symbol
+/// index. Covers `fn main` (Rust), `def main` (Python), `func main()`
+/// (Go), and `function main()` (JS/TS).
+fn detect_main_symbols(symbols: &HashMap<String, FileSymbols>, out: &mut Vec<EntryPoint>) {
+    for (path, file_syms) in symbols {
+        for def in &file_syms.definitions {
+            if def.name == "main" {
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-collectors/src/entry_points.rs]: Fixed in c89c690 by being honest in the docs rather than expanding scope. Detecting `if __name__ == \"__main__\":` requires top-level `If`-statement analysis that the current symbol index doesn't extract. Added an explicit "Out of scope (v1)" section to the module doc + EntryPoint doc + detect_main_symbols doc explaining why. Issue tracker can pick this up as a follow-up if it becomes important.

File: crates/analyzer-core/src/types.rs [hunk 4]
```diff
@@ -261,6 +261,40 @@ pub struct LanguageInfo {
     pub percentage: f64,
 }
 
+/// One discovered entry point - a place where the program can start running.
+///
+/// Surfaces both manifest-declared entry points (`Cargo.toml [[bin]]`,
+/// `package.json bin`/`scripts`, `pyproject.toml [project.scripts]`) and
+/// AST-derived ones (`fn main`, `def main`, `func main()`,
+/// `if __name__ == "__main__":`). Lets agents and contributors find every
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-core/src/types.rs]: Fixed in c89c690 — updated the EntryPoint docstring to explicitly state `__main__` blocks are NOT detected (with the reason), so docs and implementation now match.

File: crates/analyzer-collectors/src/entry_points.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-collectors/src/entry_points.rs]: Fixed in c89c690 (same change as the duplicate medium-priority comment above) — `__main__` guard detection moved to explicit out-of-scope list rather than misleadingly claimed in docs.

[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-collectors/src/entry_points.rs]: Fixed in f9f79f4 — filter `def.kind == SymbolKind::Function` so a struct/const/module literally named `main` no longer surfaces as an EntryPointKind::Main. New test `ast_main_only_matches_functions_not_other_symbol_kinds` covers the Struct + Constant case.

File: crates/analyzer-collectors/src/entry_points.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-collectors/src/entry_points.rs]: Same finding as the original review thread (matches `original_commit_id 8424963`); already addressed in c89c690 — see resolved reply on the prior thread.

File: crates/analyzer-collectors/src/entry_points.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-collectors/src/entry_points.rs]: Same finding as the original review thread (matches `original_commit_id 8424963`); already addressed in c89c690 — see resolved reply on the prior thread.

File: crates/analyzer-collectors/src/entry_points.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-collectors/src/entry_points.rs]: Same finding as the original review thread (matches `original_commit_id 8424963`); already addressed in c89c690 — see resolved reply on the prior thread.

File: crates/analyzer-core/src/types.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/analyzer-core/src/types.rs]: Same finding as the original review thread (matches `original_commit_id 8424963`); already addressed in c89c690 — see resolved reply on the prior thread.
